src/store/products/actions.ts

```diff
@@ -1 +1,21 @@
+import { AppDispatch } from "..";
 import { ProductsTypes } from "../types/products";
+
+export function getAllProducts() {
+  return async (dispatch: AppDispatch) => {
+    dispatch({ type: ProductsTypes.START_LOADING_PRODUCTS });
+    const response = await fetch(
+      "https://645b7a19a8f9e4d6e7699f11.mockapi.io/api/products"
```

Comment:
base url -"https://645b7a19a8f9e4d6e7699f11.mockapi.io/api/" should be as variable